cvmfs/swissknife_pull.cc

```diff
@@ -486,7 +501,7 @@ int swissknife::CommandPull::Main(const swissknife::ArgumentList &args) {
   }
 
   // Fetch tag list
-  if (!ensemble.manifest->history().IsNull()) {
+  if (!ensemble.manifest->history().IsNull() && !preload_cache) {
```

Comment:
Perhaps add a comment here.  For now it is the right call to not load the tag list when preloading the cache.  Eventually, however, the user should be able to decide.  It might be wanted behavior to load the full repository history.